addons/mail/static/src/core/web/thread_service_patch.js

```diff
@@ -245,6 +245,12 @@ patch(ThreadService.prototype, "mail/core/web", {
         }
         this._super(thread, replaceNewMessageChatWindow);
     },
+    removeChatWindow(channel) {
```

Comment:
Shouldn't it override `remove` like in 16.3?
Normally nothing changed in that regard.

Reply:
In 16.4 we close a `chatWindow` it also notifies the server it's state. But, if we keep the code same as 16.3 when another users will get the `channel/delete` notification they will also close the chatwindow and make and RPC call to `channel_fold` which is useless as the channel itself has been unlinked.

And it was also failing some test cases.

Reply:
And what else is that `chatWindow` should be closed before you have left the channel. So, we need to reintroduce `leaveChannel`.

Reply:
It's fine to add `{ notifyState: false }` in the leave/remove method, I updated the code with that.

If the chat window must be closed before, feel free to make a new PR to inverse the order of the calls between remove and RPC inside `leaveChannel` itself. Maybe that will fix the bug I noticed in 16.3